Merge Webdriverio into react-branch

tommyv1987 opened the pull request:
**Webdriverio**
- Base implementation of Webdriverio UI tests for explorer
- POM
- wdio.config set up and ready to be run in GH actons
- data-testids added to relevant components

mmsinclair committed:
network-explorer: initial commit

Commit:
init commit - worldMap work babelrc and some file architecture setup

Commit:
basic routing added

Commit:
routing fix

Commit:
very basic nav links added

Commit:
scaffolding for pages/routing no content yet

Commit:
changes to scaffolding and implementation of basic navbar with animation

Commit:
added basics of topmenu

Commit:
scaffolding for units not working with ts yet

Commit:
fixed useStyle issue but broke theme for fontSize

Commit:
fixed themeProvider and named export updates

Commit:
fixed hamburger menu icon

Commit:
added theme to much of Nav

Commit:
basic nav drawer working

Commit:
MUI components and SVG added

Commit:
changed mui style pkg

Commit:
now pulling down theme correctly not using makeStyles syntax

Commit:
navbar routing correctly

Commit:
fixed styled component

Commit:
dynamic colour for isSelected

Commit:
corrected type names

Commit:
wrapped nav for routing and positioning of main section and dynamic colour for selected section

Commit:
overview info panes added

Commit:
quick refactor break out components separately

Commit:
WorldMap implemented but not data

Commit:
map changed and updated to shades correctly

Commit:
live data in cards

Commit:
added any types for react simple map

Commit:
nested routing added but needs tidying and types refactored

Commit:
added tooltip to worldmap

Commit:
worldmap killed unused props

Commit:
updated MUI version to stable v5

Commit:
dark mode and ContentCard refactor complete

Commit:
refactor of DarkMode context and API into class and context setup

Commit:
context refactor for multiple APIs at top level

Commit:
mui typography used for error msging instead of jsx/html

Commit:
added typeDefs for node api types

Commit:
small changes to sx styling

Commit:
added types for api responses and main context

Commit:
promiseAll for better error handling of individual async calls

Commit:
switch out to live API for country nodes

Commit:
removal of unnecessary type any and shortening sx style block

Commit:
routing and basic mixnodes table and linking

Commit:
fixed TS error handling and ts exclude files

Commit:
refactor of class API fetch reqs

Commit:
renaming to more appropriate explorer-api

Commit:
broken - passing to Fouad

Commit:
fix for types in context main

Commit:
mixnode detail page

Commit:
rebasing back before fetch mixnode by ID was implemented

Commit:
added basic cache for huge dump of mixnode data

Commit:
broken mixnodes context

Commit:
fixed mixnode detail fetch

Commit:
added hardcoded BondBreakdown section

Commit:
added 2 col table for detail page and small refactor of ApiState type for consistent use throughout app

Commit:
basic chart with basic dark/theme implemented - no live data

Commit:
added scrollToTop useRef for Detail page

Commit:
tidied grid items

Commit:
media qry for smaller screens

Commit:
small changes

Commit:
added live data to bonds breakdown 1/2

Commit:
small changes/tweaks

Commit:
Bondbreakdown retrieves live data

Commit:
mixnode stats using live data

Commit:
added node status live data

Commit:
uptime story added with live data

Commit:
date formatting added

Commit:
mixnode map

Commit:
error handling for mixnode stats

Commit:
error handling for port stats

Commit:
improved error handling for table - unfinished

Commit:
error handling for mixnode table

Commit:
handle Loading state for 2colSmallTable

Commit:
Uptime story loading handling

fmtabbara committed:
set up data grid component

fmtabbara committed:
remove mixnode value check as handled inside MixnodesDataGrid component

fmtabbara committed:
use loading prop in data grid component

fmtabbara committed:
undo unintentional code formatting

Commit:
map blur and linkable data-grid added

Commit:
getting ready for gateways and removing con logs

Commit:
quickfix for map blur

Commit:
PR comment changes

Commit:
refactored data grid for reuseability

Commit:
Link to open Big Dipper for Blocks

Commit:
passing element to title instead of string for routing to Big Dipper

Commit:
quick fix for element passed as title for contentCard

Commit:
fix for colour coding nodes

Commit:
nuts and bolts of search and results per page are working

Commit:
media query for responsive search and no-per-page toolbar

Commit:
broke out search and pagesize to separate toolbar

Commit:
fix for going back to mixnodes datagrid and refetching

Commit:
corrected typings for WorldMap

Commit:
removed API for topojson

Commit:
Cleaner implementation of formatting inline for datagrid

Commit:
added Type to Datagrid Rows for mixnode

Commit:
removed optional from type for Datagrid

Commit:
added page listing the Gateway nodes

Commit:
adding clickable location to handleSearch

Commit:
tidying util functions and removing dead useEffect

Commit:
Validators link to Big Dipper

Commit:
added validators link to side nav as per Issues card

Commit:
SVG icons

Commit:
PR tweak to move logic to routes

mmsinclair committed:
Add missing constant

Commit:
removed dead code post rebase

Commit:
fixed light dark mode for DataGrid

Commit:
light dark mode works on SVGs in Nav

Commit:
moving logic back to Nav to avoid window object issues

Commit:
neater ternary for SVG icons dark mode

Commit:
Better Linking/Styling for cells

Commit:
corrected prop/attr name in svg to Reactify it

Commit:
moved api url to constants

Commit:
SVGs dark now governed by context not props so reverting renderIcon method back to key value setup

Aid19801 committed:
Merge pull request #812 from nymtech/validators-link

Validators link to Big Dipper

Commit:
percentage for bond total added

Commit:
SVGs for Overview cards Mixnodes Gateways and Validators

Commit:
decimalised formatted punks and % of bond for BondBreakdown card

Commit:
number formatting via validator module

Commit:
adding cossmjs math pkg

Commit:
unfinished refactor BondBreakdown

Commit:
first few ui tweaks

Commit:
resolved git conflicts

Commit:
Adding google font Open Sans as per designs

Commit:
DataGrid unstylable in theme so nuking in css

Commit:
adding theming to Block Height card not hardcoded colours

Commit:
DataGrid styling

Commit:
Nav styling colours but without hover fix

Commit:
theme for bond breakdown

Commit:
killing con logs

Commit:
Datagrid styling

Commit:
Nav bar working

Commit:
added lines to nav

Commit:
removed cursive from fallback fonts

Commit:
trimming and refactoring

Commit:
removed dead isActive code from nav options

Commit:
Color correction for theme on 2col table

Commit:
Moved cell styles out to UniversalDatagrid for reuseability

Commit:
Nav colors moved to theme

Commit:
Removing comments and dead code

Commit:
DataGrid UI improvements

Commit:
theming for Overview content card

Commit:
Bonds updated from UPUNK to PUNK

Commit:
corrected SVG warning on stroke-width

Commit:
corrected SVG warning on stroke-width

Commit:
added Boolean class instead of ternary

Commit:
added Boolean class instead of ternary

Commit:
fixing up svg attr to jsx props

Commit:
merging UPUNK changes into ui-tweaks

Commit:
last instance of Boolean

Commit:
BondBreakdown handles 0 delegations

Commit:
formatting for webpack config and svgs

Commit:
resolved merge conflicts

mmsinclair committed:
Merge pull request #821 from nymtech/punk-formatting

UPUNK => PUNK Formatting via Nym Validator Client

mmsinclair committed:
Merge pull request #830 from nymtech/ui-tweaks

UI tweaks

mmsinclair committed:
Add `npm run lint` and `npm run lint:fix` targets to `package.json`

mmsinclair committed:
Allow `.vscode` directories - exclude them individually like has been done already in the `.gitignore` directory

mmsinclair committed:
Add `vscode` action to run `eslint` on save for the `/explorer/**` sub-directory

mmsinclair committed:
eslint auto fix

mmsinclair committed:
Fix some easy eslint issues

Commit:
removing grid pipes and pastel map colors

Commit:
Grid xl lg values to align with Search Toolbar

mmsinclair committed:
Merge pull request #834 from nymtech/ui-tweaks-2

3 new UI tweaks

mmsinclair committed:
Merge remote-tracking branch 'origin/develop' into feature/network-explorer-react

mmsinclair committed:
GitHub Actions: do not trigger Rust actions when the paths are only `/explorer/**`

See https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onpushpull_requestpaths for details.

mmsinclair committed:
GitHub Actions: run eslint and annotate pull requests

Commit:
socials added to Appbar and Footer

Commit:
smaller darkmode icon for mobile

Commit:
cleaner code for nav dark light selector

Commit:
almost all lint fixes

Commit:
post linting Nav fix

Commit:
killed con log and removed unused dep

Commit:
ref type and removed 1x ts ignore from worldMap

Commit:
disabl nested tern w/ nav is refactored on diff br

Commit:
icons smaller put into mui List format

Commit:
Added hover effect to match DarkLight switch

Commit:
bringing linting and social icons changes in

Commit:
ts ignore for worldMap vs no ts decl

Aid19801 committed:
Merge pull request #839 from nymtech/linting-fixes

Linting fixes

Commit:
merging linting changes

Commit:
parking changes

Commit:
Flipped to MUI SVGs

Commit:
re-added external links to Socials

Commit:
Merge branch 'social-icons' into nav-changes

Commit:
nav functionality working

Commit:
spacing on Mixnode detail page

Commit:
datagrid alignment & detail page spacing

Commit:
map but no datagrid yet

Commit:
killed old SVGs now using mui icons

Commit:
added palette instead of strings

Commit:
Mixnode Map page working still needs tidying

Commit:
better lg xl responsive on Overview and sanitized Page Titles

Commit:
removed typography from imports as unused now

Commit:
search, sort working & added LG XL responsiveness

Commit:
Routing root reqs direct to Overview

Commit:
basic 404 page and btn back to overview

Commit:
killed fragments and comments

Commit:
updated Bond total in column

Commit:
Change bond col to type number not text

Commit:
Added field to DataGrid and updated MixnodeToGridrow logic

Commit:
Added type number so sorting works properly.

Commit:
added %self to Detail page

Commit:
basic scroll working desktop

Commit:
delegations now popout and scroll according to designs

Commit:
added stickyHeader and killed dead code

Commit:
ExpandMore only renders if delegations exist

Commit:
killed old svg icons

Commit:
added theme to Overview SVGs

Aid19801 committed:
Merge pull request #838 from nymtech/social-icons

Social Icons & Dark Mode selector

Aid19801 committed:
Merge pull request #844 from nymtech/nav-changes

Nav changes

Aid19801 committed:
Merge branch 'feature/network-explorer-react' into mixnode-detail-alignment

Aid19801 committed:
Merge pull request #845 from nymtech/mixnode-detail-alignment

Mixnode Detail Alignment

Aid19801 committed:
Merge pull request #848 from nymtech/equalise-spacing-pages

Equalise spacing pages

Aid19801 committed:
Merge branch 'feature/network-explorer-react' into node-map

Aid19801 committed:
Merge pull request #849 from nymtech/node-map

Mixnode-map page

Aid19801 committed:
Merge pull request #850 from nymtech/route-to-overview

Routing root reqs direct to Overview

Aid19801 committed:
Merge pull request #851 from nymtech/add-404-page

basic 404 page and btn back to overview

Aid19801 committed:
Merge branch 'feature/network-explorer-react' into bug-bond-column

Aid19801 committed:
Merge pull request #852 from nymtech/bug-bond-column

Bug bond column

Aid19801 committed:
Merge branch 'feature/network-explorer-react' into percent-self

Aid19801 committed:
Merge pull request #853 from nymtech/percent-self

Add Percent Self to Mixnodes Datagrid

Commit:
Merge branch 'feature/network-explorer-react' into scrollable-delegators

Commit:
bringing Title into other pages

Aid19801 committed:
Merge pull request #859 from nymtech/scrollable-delegators

Scrollable delegators

Commit:
linting fix

Aid19801 committed:
Merge pull request #863 from nymtech/merge-fixes

Merge Fixes

Commit:
pagination and spacing of gateways cols

Commit:
linting fix

tommyv1987 committed:
[WIP] first commit for nym-explorer tests

. POM
. Basic test navigating to explorer overview page
. Lots more to come

tommyv1987 committed:
typo

Commit:
style override for pagination

tommyv1987 committed:
update path for unit tests

Commit:
added hamburger and changed appbar to fixed

Commit:
bringing in other lint fix to pass linting

tommyv1987 committed:
Updating data-test-ids to identify elements

Add Id's to select components on the explorer

tommyv1987 committed:
WIP 

temporarily running the test against localhost to cater for the changes with test ids

tommyv1987 committed:
Add configuration urls to run across different envs

. Now configurable by setting the ENV: variable in the scripts
. Can run the scripts by `yarn test:xxx`
. More to do here for running in CI

Aid19801 committed:
Merge pull request #866 from nymtech/add-pagination

pagination and spacing of gateways cols

Commit:
PR feedback changes

Aid19801 committed:
Merge pull request #867 from nymtech/nym-logo-link

Hamburger Menu & Fixed AppBar

tommyv1987 committed:
react-branch

tommyv1987 committed:
Update config file for running in GH actions

. Still WIP

Aid19801 commented on explorer/src/tests/webdriver/tests/pageobjects/overview.page.js:
```diff
@@ -0,0 +1,41 @@
+    }
+}
+
+export default new Overview()
```
Unsure what's been discussed re Webdriver but was there a reason we're not writing these in TS?

tommyv1987 replied:
Noted, will change the project to TS

mmsinclair replied:
Yes, it would be best if they are in Typescript

Aid19801 commented on explorer/src/tests/webdriver/wdio.conf.js:
```diff
@@ -0,0 +1,190 @@
+    capabilities: [{
+        maxInstances: 5,
+        //
+        browserName: 'chrome',
```
Might be overkill given so much is MUI & heavily tested already, but is it possible to add other browsers to this? Firefox? Edge?

tommyv1987 replied:
Sure, it's all configurable. There's another config file that will be run specifically for Github actions. This was just the boilerplate out the box on set up.

tommyv1987 committed:
TS - Implementation

. Moving to TS 
. Few issues running locally at the moment will resolve shortly after, checking in branch to keep update
. Need to investigate the tsconfig.json
. Everything is a lot cleaner
. WIP

tommyv1987 committed:
Add type

tommyv1987 commented:
Code is now elsewhere closing.
